indexers/social_feed/social_feed.js

```diff
@@ -241,6 +243,82 @@ async function getBlock(block: Block) {
     }
   }
 
+    async function handlePromotion(
+    accountId,
+    blockHeight,
+    blockTimestamp,
+    receiptId,
+    promoteString
+  ) {
+    const promotion = JSON.parse(promoteString);
+    console.log("Promotion", promotion);
+    const promotionOperation = promotion.value.operation;
+
+    if (promotionOperation === "add") {
+      console.log("handling add promotion");
+      await _handleAddPromotion(
+        promotion,
+        accountId,
+        blockHeight,
+        blockTimestamp,
+        receiptId
+      );
+      return;
+    } else {
+      // if an operation is implemented, we can handle it here
+      console.log("Operation not implemented");
+    }
+  }
+
+    async function _handleAddPromotion(
+    promotion,
+    accountId,
+    blockHeight,
+    blockTimestamp,
+    receiptId
+  ) {
+    // Add your code here
+    const postAuthor = promotion.value.post.path.split("/")[0];
+    const postBlockHeight = promotion.value.post.blockHeight;
+    const promotionType = promotion.value.type;
+
+    console.log("Post Author", postAuthor);
+    console.log("Post Block Height", postBlockHeight);
+    console.log("Promotion Type", promotionType);
+    try {
+      const posts = await context.db.Posts.select(
+        { account_id: postAuthor, block_height: postBlockHeight },
+        1
+      );
+
+      if (posts.length > 0) {
+        const post = posts[0];
+        let content = JSON.parse(post.content);
+
+        console.log("Post found in database", post);
+        console.log("Post content", content);
+
+        delete promotion["item"];
+
+        const promotionData = {
+          account_id: accountId,
+          receipt_id: receiptId,
+          block_height: blockHeight,
+          block_timestamp: blockTimestamp,
+          promotion_type: promotionType,
+          post_id: post.id,
+        };
+
+        // Call GraphQL mutation to insert a new promotion
+        await context.db.Promote.insert(promotionData);
+
+        console.log(`Promotion by ${accountId} has been added to the database`);
+      }
+    } catch (e) {
+      console.log("Error handling add promotion", JSON.stringify(e));
```

Comment:
noted that this file has an existing pattern of logging errors as `log` instead of `error`. @gabehamilton where do these logs output? Should we improve the logging by writing errors as `error`?